onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -1488,6 +1492,15 @@ Status PropagateCastOps::ApplyImpl(Graph& graph, bool& modified, int graph_level
       }
     }
 
+    // Eliminate FP32 input casts and FP16 output casts
+    for (auto node_index : node_topology_list) {
+      Node* node = graph.GetNode(node_index);
+      if (nullptr != node &&
+          std::find(removed_nodes.begin(), removed_nodes.end(), node->Index()) == removed_nodes.end()) {
+        local_modified |= RemoveInputOutputUpDownCasts(graph, node, removed_nodes, level_, logger);
```

Comment:
Do we still need to call this function "RemoveInputOutputUpDownCasts" ? 
My impression is that, this overlaps with existing InsertAndReduce implementation. 

What are the scenarios this  RemoveInputOutputUpDownCasts covers?

Reply:
It will cover scenario when the opcode is not in fp16_allowed list. As per our conversation. I am removing this function.